tests/src/cgeo/geocaching/location/GeoPointParserTest.java

```diff
@@ -139,6 +137,264 @@ public static void testFloatingPointNbsp() {
         assertGeopointEquals(GeopointParser.parse("47.648883  122.348067\u00a0"), GeopointParser.parse("N 47° 38.933 E 122° 20.884"), 1e-4f);
     }
 
+    public static void testDoubleComma() {
+        assertThat(GeopointParser.parse("47.648883,122.348067").equals(GeopointParser.parse("47.648883 122.348067")));
+    }
+
+    public static void testGerman() {
+        assertThat(GeopointParser.parse("N 47° 38.933 O 122° 20.884")).isEqualTo(GeopointParser.parse("N 47° 38.933 E 122° 20.884"));
+    }
+
+    public static void testNoSpace() {
+        assertThat(GeopointParser.parse("N47°38.933E122°20.884")).isEqualTo(GeopointParser.parse("N 47° 38.933 E 122° 20.884"));
+    }
+
+    public static void testUTM() {
+        GeopointParser.parse("54S E 293848 N 3915114");
+    }
+
+    public static void testZero() {
+        GeopointParser.parse("00° 00.000′ 000° 00.00′");
+        GeopointParser.parse("00° 00′ 00.00″ 000° 00′ 00.00″");
+        GeopointParser.parse("00° 00′ 000° 00′");
+        GeopointParser.parse("00° E 000°");
+        try {
+            GeopointParser.parse("00° 00.001′ 000° 00.01′");
+            fail();
+        } catch (final Geopoint.ParseException e) {
+            // expected
+        }
+        try {
+            GeopointParser.parse("00° 00′ 00.01″ 000° 00′ 00.01″");
+            fail();
+        } catch (final Geopoint.ParseException e) {
+            // expected
+        }
+        try {
+            GeopointParser.parse("00° 01′ 000° 01′");
+            fail();
+        } catch (final Geopoint.ParseException e) {
+            // expected
+        }
+        try {
+            GeopointParser.parse("01° E 000°");
+            fail();
+        } catch (final Geopoint.ParseException e) {
+            // expected
+        }
+    }
+
+    public static void testFormula() {
+        try {
```

Comment:
These expected failures cause a lot of duplicate code blocks. Maybe extract into a `assertParsingFails(final String geopointToParse)` method.